content/React Query and management of server state.md

```diff
@@ -0,0 +1,75 @@
+- Inactive: React Query has a garbage collector for managing cache on the browser, this state in some manner indicates on React query that if the data is not be using on the application then it's gonna be potentially deleted after some time. This is a great feature of React Query, because we want to persist the recent information for not fetching data all the time and improve the speed/UX of our interfaces.
+- Deleted: This happens when the data was inactive for a determinate time, in which case is deleted from the cache. This time can be configurable locally for each query or globally.
+
+With this approach, React Query handles the asynchronous information of our application clearly, allow us to scale and maintain our code in a better way.
+
+## Introduction to React query API
+Although React Query has only two hooks, it is highly configurable in all aspects, since the retry delay time of the queries until what is the interval time of re-fetch in background. 
+But let's start with the most basic API that is well documented in [their repository](https://github.com/tannerlinsley/react-query)
+``` javascript 
+const { status, data, error } =  useQuery('movies', fetchTodoList);
+```
+This hook provided us the status of the fetch(loading, error, or success), data and error if are defined. Up to this point is a pretty normal hook for fetching information, but as we see the query is related to a key(in this case `movies`), this is a unique-global key that is used for associate the query information between the components, in this way we can re-use this hook use the same information in anywhere of the DOM tree. If you follow the classic redux action-reducer cycle for fetching data, **this hooks it will save up a lot of repetitive code**.
+
+Among other things, React Query also allows us to filter the information with a simple system.
+```javascript
+// An individual movie
+useQuery(['movies', 5], ...);
+
+// A list of movies that are "released"
+useQuery(['movies', { type:  'released' }], ...);
+```
+And also, for contemplate the case that the client-side alter the server state React Query introduce the concept of **mutations**(well known for GraphQL developers), let's make a quick example
+
+```javascript
+const [mutate, mutationState] = useMutation(patchMovies, {
+   onSuccess: newMovie => {
+     // Update `movies` while refetching all the list on background
+     queryCache.refetchQueries(movies");
+     queryCache.setQueryData(["movies", { id: newMovie.id }], newMovie);
+   }
+ });
+
+ const onSave = newMovie => mutate(newMovie);
+```
+With `onSave` function we're re-fetching the information on the background while immediately change the specific information(in this case the new movie) in the user interface.
+
+## Manage cached data
+Save data that we fetched and how we saved is an important aspect of our application, improves the sensation of speed in our interfaces, and avoiding hit the backend all the time.
```

Comment:
`_Saving_ data that we fetched and how we _save it_ (?) is an important aspect of our application, _it_ improves the sensation of speed in our interfaces, and _avoids hitting_ the backend all the time.`